mlflow/store/tracking/abstract_store.py

```diff
@@ -219,6 +225,15 @@ def log_metric(self, run_id, metric):
         """
         self.log_batch(run_id, metrics=[metric], params=[], tags=[])
 
+    def log_metric_async(self, run_id, metric):
```

Comment:
high level question - do we want to create `log_metric_async` or add a flag `async` to existing `log_metric`? Either works for me, but I think we should be consistent with fluent and client api for easier maintenance.